sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/authority/AuthorityRuleChecker.java

```diff
@@ -29,40 +29,28 @@
 
     static boolean passCheck(AuthorityRule rule, Context context) {
         String requester = context.getOrigin();
+        boolean contain = false;
 
         // Empty origin or empty limitApp will pass.
         if (StringUtil.isEmpty(requester) || StringUtil.isEmpty(rule.getLimitApp())) {
             return true;
         }
 
         // Do exact match with origin name.
-        int pos = rule.getLimitApp().indexOf(requester);
-        boolean contain = pos > -1;
+        contain = rule.getLimitApp().contains(requester);
```

Comment:
If limitApp is `foobar,test`, and requester is ‘foo’， ‘foo’ will be blocked mistakely.  So we need remain the double check.  In most cases the requester will be passed because the first check. The loop second check confirms that requester never be blocked mistakely.

Reply:
Oh, I really didn't take that into consideration. Thank you for your correction，
Now That I've fixed it, the performance is really down, and after 100 million calls, it's three times faster